pkg/sqlite/configmap.go

```diff
@@ -0,0 +1,162 @@
+package sqlite
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/ghodss/yaml"
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
+	"github.com/operator-framework/operator-registry/pkg/registry"
+	"github.com/sirupsen/logrus"
+	"k8s.io/api/core/v1"
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+const (
+	ConfigMapCRDName     = "customResourceDefinitions"
+	ConfigMapCSVName     = "clusterServiceVersions"
+	ConfigMapPackageName = "packages"
+)
+
+// ConfigMapLoader loads a configmap of resources into the database
+// entries under "customResourceDefinitions" will be parsed as CRDs
+// entries under "clusterServiceVersions"  will be parsed as CSVs
+// entries under "packages" will be parsed as Packages
+type ConfigMapLoader struct {
+	store     registry.Load
+	configMap v1.ConfigMap
+	crds      map[registry.APIKey]*unstructured.Unstructured
+}
+
+var _ SQLPopulator = &ConfigMapLoader{}
+
+func NewSQLLoaderForConfigMap(store registry.Load, configMap v1.ConfigMap) *ConfigMapLoader {
```

Comment:
Wouldn't it be better to remove the intermediate `ConfigMap` and just consume a directory of manifests? Since the `ConfigMap` will probably be too large to `kubectl create` anyway?

Reply:
That's a decision we can make on the OLM side (what to do with the configmap after we have them converted here). I was thinking that it would make sense to continue to support the configmaps as a low barrier to entry; not really convinced they're that much easier though.